File: src/openprecedent/schemas/case.py
from datetime import datetime
from enum import StrEnum

from pydantic import BaseModel, ConfigDict


class CaseStatus(StrEnum):
    STARTED = "started"
    COMPLETED = "completed"
    FAILED = "failed"
    PAUSED = "paused"


class Case(BaseModel):
    model_config = ConfigDict(extra="forbid")

    case_id: str
    title: str
    status: CaseStatus
    user_id: str | None = None
    agent_id: str | None = None
    started_at: datetime
    ended_at: datetime | None = None
    final_summary: str | None = None

File: rust/openprecedent-contracts/src/case.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use strum::{Display, EnumString};

#[derive(Clone, Copy, Debug, Eq, PartialEq, Serialize, Deserialize, Display, EnumString)]
#[serde(rename_all = "snake_case")]
#[strum(serialize_all = "snake_case")]
pub enum CaseStatus {
    Started,
    Completed,
    Failed,
    Paused,
}

#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
pub struct Case {
    pub case_id: String,
    pub title: String,
    pub status: CaseStatus,
    pub user_id: Option<String>,
    pub agent_id: Option<String>,
    pub started_at: DateTime<Utc>,
    pub ended_at: Option<DateTime<Utc>>,
    pub final_summary: Option<String>,
}
